pyrevolve/pyrevolve.py

```diff
@@ -56,55 +58,150 @@ class BaseRevolver(object):
         * Find a better name than `checkpoint`, as the object with that name
           stores live data rather than one of the checkpoints.
     """
+
     __metaclass__ = ABCMeta
 
-    def __init__(self, checkpoint, fwd_operator, rev_operator,
-                 n_checkpoints=None, n_timesteps=None, timings=None,
-                 profiler=None):
-        """Initialise checkpointer for a given forward- and reverse operator, a
-        given number of time steps, and a given storage strategy. The number of
-        time steps must currently be provided explicitly, and the storage must
-        be the single-staged memory storage."""
-        if(n_timesteps is None):
-            raise Exception("Online checkpointing not yet supported. Specify \
-                              number of time steps!")
-        if(n_checkpoints is None):
-            self.n_checkpoints = cr.adjust(n_timesteps)
-        else:
-            self.n_checkpoints = n_checkpoints
+    storage_list = []
+
+    def __init__(
+        self,
+        checkpoint,
+        fwd_operator,
+        rev_operator,
+        n_checkpoints,
+        n_timesteps,
+        storage_list=[],
```

Comment:
This is prone to a bug as given [here](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument). Should be `storage_list=None` here and `if storage_list is None: storage_list = []` in the function body somewhere.

Reply:
Ok